info-compiler/src/test/java/com/google/sps/InfoCompilerTest.java

```diff
@@ -0,0 +1,123 @@
+      " \"elections\": [" +
+      "  {" +
+      "   \"id\": \"2000\"," +
+      "   \"name\": \"VIP Test Election\"," +
+      "   \"electionDay\": \"2013-06-06\"" +
+      "  }" +
+      " ]" +
+      "}";
+  private JsonObject JSON;
+  private InfoCompiler infoCompiler;
+
+  @Before
+  public void initialize() {
+    infoCompiler = new InfoCompiler();
+    JSON = new JsonObject();
+    JSON.addProperty("kind", "civicinfo#electionsqueryresponse");
+    JsonArray elections = new JsonArray(1);
+    JsonObject election = new JsonObject();
+    election.addProperty("id", "2000");
+    election.addProperty("name", "VIP Test Election");
+    election.addProperty("electionDay", "2013-06-06");
+    elections.add(election);
+    JSON.add("elections", elections);
+  }
+
+  @Test
+  public void queryCivicInformation() throws Exception {
+    // Query the Civic Information API with a mock HTTP client + mock callback function of type
+    // {@code ResponseHandler<String>} that converts the {@code HttpResponse} response to
+    // {@code RESPONSE}, and subsequently convert {@code RESPONSE} to {@code json}.
+    // Since String {@code RESPONSE} and JsonObject {@code JSON} match in content, {@code json}
+    // should be exactly the same as {@code JSON}.
+    CloseableHttpClient httpclient = mock(CloseableHttpClient.class);
+    ArgumentCaptor<ResponseHandler<String>> argumentCaptor =
+        ArgumentCaptor.forClass(ResponseHandler.class);
+    when(httpclient.execute(anyObject(), argumentCaptor.capture())).thenReturn(RESPONSE);
+    JsonObject json = infoCompiler.queryCivicInformation(ELECTION_QUERY_URL, httpclient);
+    Assert.assertEquals(json, JSON);
+  }
+
+  // @TODO [Currently there are not good ways to mock {@code Datastore} (1) objects or (2)
+  // operations, as indicated in {@code WebCrawlerTest}'s {@code storeInDatabase_*()} tests.
+  // Might explore more advanced ways of mocking/spying to unit-test {@code Datastore} related
+  // steps in future development.]
+
+  @Test
+  public void queryAndStoreBaseElectionInfo() {
+    // Execute the entire process of querying and storing base election information. Should execute
+    // without exceptions and the database should be populated correctly. This confirms the validity
+    // of the query URLs and HTTP requests to the Civic Information API.
+    // This is an integrated test.
+    infoCompiler.queryAndStoreBaseElectionInfo();
+  }
+
+  @Test
+  public void queryAndStoreElectionContestInfo() {
+    // Execute the entire process of querying and storing election/position/candidate information.
+    // This step is intended to run right after {@code queryAndStoreBaseElectionInfo()}.
+    // Should execute without exceptions and the database should be populated correctly. This
+    // confirms the validity of the query URLs and HTTP requests to the Civic Information API.
+    // This is an integrated test.
+    infoCompiler.queryAndStoreElectionContestInfo();
```

Comment:
You're not really testing anything here. It doesn't make sense to have a test without assertions. You should ideally test that it succeeds with valid input or fails with bad data. To do so, you really do need to set up a mock datastore instance. Until that's figured out I would rather you leave the tests empty with a TODO so as to avoid the impression that this code has test coverage

Reply:
Will do!

Reply:
I re-wrote Datastore unit tests using Datastore emulators.